src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -161,7 +167,8 @@ func (enc *TimestampEncoder) writeAnnotation(stream encoding.OStream, ant ts.Ann
 
 	stream.WriteBytes(buf[:annotationLength])
 	stream.WriteBytes(ant)
-	enc.PrevAnnotation = ant
+
+	enc.PrevAnnotationChecksum = checksum
```

Comment:
nit: probably cleaner for the checksum to be computed here instead of `shouldWriteAnnotation`

Reply:
Yeah, but I did not want to compute it twice (as this is performance sensitive code).

Reply:
Ah yeah guess you need to do the comparison